### ISSUE TITLE
mdbx implementation tightly coupled with Reth's Tables

### ISSUE THREAD

**Issue by Vid201** [paradigmxyz/reth#4546]
### Describe the bug

There are multiple checks like this:

`let table = Tables::from_str(T::NAME).expect("Requested table should be part of `Tables`.");`

in the mdbx implementation, making mdbx impossible to use in your project with different tables.

This was possible before commit `019ddd1edca70f8af4de8aa6447c29acbc46d836`.

### Steps to reproduce

Source: https://github.com/paradigmxyz/reth/blob/main/crates/storage/db/src/implementation/mdbx/tx.rs#L43

**Comment by Vid201** [comment]
> @Vid201 hey, I'm not quite sure if `reth-db` crate is intended to be used with databases other than Reth's one, for that you might want to check out `reth-libmdbx` instead which doesn't have any reth-related dependencies.
> 
> What is your use-case, is it Reth-related project with new tables, or a completely separate thing and you just use our MDBX abstractions?

Hey @shekhirin , hmm,  I can look into `reth-libmdbx` , but a lot of code from `reth-db` would have to be duplicated probably.

We are using that in ERC-4337 bundler Silius for user operation mempool implementation (https://github.com/silius-rs/silius/tree/main/crates/uopool/src/database). At the moment, tables are different than in Reth, but we might use some of the Reth tables in the future.

**Comment by shekhirin** [comment]
@Vid201 makes sense, I agree. Given we also want to be able to extend Reth with custom stages on CLI builder level, it's actually a must for us too :)

We will support that, thanks for flagging!
